### PR TITLE
[VUFIND-1665] Improve link templates across all backends.

### PR DESCRIPTION
**Pull request by demiankatz**
This is a follow-up to #3834, making link templates less redundant and extending consistent limit support across all backends.

I can't easily test Pazpar2, Primo or Summon, but since I can test WorldCat, and the other changes are basically identical, I feel reasonably confident that this is all good. If @EreMaijala can try Primo links and anyone with spare time can double-check that I got all my details right, all available help is appreciated. :-)

TODO
- [x] Resolve [VUFIND-1665](https://openlibraryfoundation.atlassian.net/browse/VUFIND-1665) when merging.
- [x] Update #2612 to match after merging.

### CODE REVIEW

File: themes/bootstrap3/templates/RecordDriver/DefaultRecord/link-generic.phtml
Comment (maccabeelevine) [inline themes/bootstrap3/templates/RecordDriver/DefaultRecord/link-generic.phtml:6]: I see that this `quoteQuery` logic accurately reproduces the current behavior, so it's reasonable to maintain the two options.  But I'm wondering whether we actually know that the "no quotes" option is needed for some types/backends, or if it may just be an oversight that was reproduced across backends and now we're supporting it as a feature.  In case someone picks up that thread in the future.

**Review by maccabeelevine — APPROVED** [review]
DefaultRecord works as expected.  I can't test the other backends either.  Code is a great improvement.

File: themes/bootstrap3/templates/RecordDriver/DefaultRecord/link-generic.phtml
Comment (demiankatz) [reply themes/bootstrap3/templates/RecordDriver/DefaultRecord/link-generic.phtml:6]: In some cases, I think it's definitely intentional -- e.g. in some backends, author search is unquoted to support inconsistent first/last name order. In other cases it seems inconsistently applied, and it's possible as you say that it's simply an oversight or error. For this PR, my intent was to keep the functionality the same to reduce potential future confusion... but as you say, it may be worth revisiting in the future.

**Comment by demiankatz** [comment]
Thanks, @maccabeelevine -- I'll wait to get an approval from @EreMaijala before merging, but hopefully this is ready to go now. :-)

File: themes/bootstrap3/templates/RecordDriver/DefaultRecord/link-generic.phtml
Comment (maccabeelevine) [reply themes/bootstrap3/templates/RecordDriver/DefaultRecord/link-generic.phtml:6]: Makes sense and good to know for the future.

**Review by EreMaijala — APPROVED** [review]

**Comment by EreMaijala** [comment]
Looks good to me, and Primo links work fine too. :)

**Pull request merged**
